tests/cephadm/test_replace_osd.py

```diff
@@ -0,0 +1,60 @@
+from ceph.ceph_admin import CephAdmin
+from ceph.ceph_admin.common import fetch_method
+from ceph.ceph_admin.daemon import Daemon
+from cli.cephadm.cephadm import CephAdm
+from utility.log import Log
+
+log = Log(__name__)
+
+
+def run(ceph_cluster, **kw):
+    config = kw["config"]
+    replace_config = config.get("replace", {})
+    add_config = config.get("add", {})
+    installer = ceph_cluster.get_nodes(role="installer")[0]
+
+    # Replace OSD
+    osd_id_list = replace_config.get("pos_args", [])
+    node_name_list = replace_config.get("nodes", [])
+    if not node_name_list or not osd_id_list or not osd_id_list[0]:
+        log.error("Missing node or OSD ID for replace OSD test")
+        return 1
+    osd_id = str(osd_id_list[0])
+    node_name = node_name_list[0]
+    cephadm = CephAdm(installer)
+    log.info("Replacing OSD {}".format(osd_id))
+    cephadm.ceph.orch.osd.rm(osd_id=osd_id, replace=True)
+    log.info("OSD {} has been replaced successfully".format(osd_id))
+
+    # Add OSD
+    node_name_list = add_config.get("pos_args", [])
+    if not node_name_list or len(node_name_list) < 2:
+        log.error("Missing node or device for add OSD test")
+        return 1
+    node_name = node_name_list[0]
+    command = add_config.get("command", [])
+    daemon = Daemon(cluster=ceph_cluster, **add_config)
+    log.info("Adding OSD on {}".format(node_name))
+    method = fetch_method(daemon, command)
+    method(add_config)
+
+    # Check if the replaced OSD has same ID
+    cephadm = CephAdmin(cluster=ceph_cluster, **config)
+
+    # Get the OSD ID of the new OSD after the replacement
+    cmd = f"ceph osd tree | grep osd.{osd_id} | cut -d' ' -f2"
```

Comment:
@msainiRedhat One suggestion, do you think its better to move this functionality to cli? I mean another .py file called osd ? So that we can remove these .shell calls

Reply:
+1 for above comment.